src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -470,16 +473,23 @@ public static PathConfig fromSourceProjectRascalManifest(ISourceLocation manifes
         IListWriter libsWriter = vf.listWriter();
         IListWriter srcsWriter = vf.listWriter();
         IListWriter messages = vf.listWriter();
-        
-        if (!projectName.equals("rascal")) {
-            // always add the standard library but not for the project named "rascal"
-            // which contains the source of the standard library
-            try {
-                libsWriter.append(resolveCurrentRascalRuntimeJar());
-            }
-            catch (IOException e) {
-                messages.append(Messages.error(e.getMessage(), manifestRoot));
-            }
+
+        // Approach:
+        //   - Rascal modules of `rascal` inside `std` are loaded from the
+        //     "current `rascal`" (as per `resolveCurrentRascalRuntimeJar`).
+        //   - Rascal modules of `rascal` outside `std` are loaded from the
+        //     "current project" of this path config (as per `manifestRoot`).
+        //   - Java classes of `rascal`, regardless of inside/outside `std`, are
+        //     loaded from the current `rascal`.
+        //
+        // Thus, Rascal modules and Java classes of `rascal` inside `std` are
+        // guaranteed to be consistently loaded from the same version.
+        try {
```

Comment:
I like this approach, less special casing of the project named "rascal". As we already had a user that named their first proejct "rascal" and ran into weird issues.

Reply:
It's possible to get the loc of the current rascal jar using `PathConfig.getCurrentRascalRuntime()`, however it's best to use `std://` since the .tpl files for the standard library will be containing those references. So for interpreter stacktraces and debugging information and such to line up with the information that the summary for the LSP uses, we should also use `std://`